tests/unit/test_client.py

```diff
@@ -1002,3 +1002,95 @@ def __call__(self):
     with_retry(FailerUntil(2).__call__)()
     with pytest.raises(SomeException):
         with_retry(FailerUntil(3).__call__)()
+
+
+def assert_headers_with_role(headers, role):
+    catalog = "test_catalog"
+    schema = "test_schema"
+    user = "test_user"
+    source = "test_source"
```

Comment:
Remove `catalog`, `schema` and `source` as it unrelates to role tests.

Reply:
:+1: